StripeIdentity/StripeIdentity/Source/NativeComponents/Coordinators/VerificationSheetController.swift

```diff
@@ -391,33 +417,34 @@ final class VerificationSheetController: VerificationSheetControllerProtocol {
         updateDataResult: Result<StripeAPI.VerificationPageData, Error>,
         completion: @escaping () -> Void
     ) {
-        guard case .success(let resultData) = updateDataResult
-        else {
-            transitionWithVerificaionPageDataResult(updateDataResult, completion: completion)
-            return
-        }
+        handleVerificationPageDataResult(collectedData: collectedData, updateDataResult: updateDataResult, completion: completion) { _ in
+            self.checkSubmitAndTransition(
+                updateDataResult: updateDataResult,
+                completion: completion
+            )
 
-        // Only merge when updateDatResult is successful and has no errors
-        if let collectedData = collectedData, resultData.requirements.errors.isEmpty {
-            self.collectedData.merge(collectedData)
         }
-
-        checkSubmitAndTransition(
-            updateDataResult: updateDataResult,
-            completion: completion
-        )
     }
 
     /// Calculate the clearData parameter from the collectedData to be generated by the following
     ///    allTypes - typesAlreadyCollected - typesToBeCollected
     private func calculateClearData(
         dataToBeCollected: StripeAPI.VerificationPageCollectedData
     ) -> StripeAPI.VerificationPageClearData {
-        return .init(
-            clearFields: Set(StripeAPI.VerificationPageFieldType.allCases).subtracting(
+
+        let initialMissings: Set<StripeAPI.VerificationPageFieldType>
+        do {
+            initialMissings = try verificationPageResponse?.get().requirements.missing ?? Set()
+        } catch {
+            assertionFailure("verificationPageResponse is nil, using StripeAPI.VerificationPageFieldType.allCases as initialMissings")
```

Comment:
this will crash in dev builds and will be a no-op in release builds. Is that intentional?

Reply:
Yes, this condition is never supposed to happen in prod, as backend always sets `verificationPageResponse?.get().requirements.missing`, just added a log for sanity check as we do similar logs in other places.